casper/src/main/scala/coop/rchain/casper/SafetyOracle.scala

```diff
@@ -269,17 +270,10 @@ sealed abstract class SafetyOracleInstances {
             if x.toString > y.toString // TODO: Order ByteString
           } yield (x, y)).toList.filterA {
             case (first: Validator, second: Validator) =>
-              // TODO: Replace with equivalent of <&&>
-              Monad[F].ifM(seesAgreement(first, second))(
-                Monad[F].ifM(seesAgreement(second, first))(
-                  Monad[F].ifM(neverEventuallySeeDisagreement(first, second))(
-                    neverEventuallySeeDisagreement(second, first),
-                    false.pure[F]
-                  ),
-                  false.pure[F]
-                ),
-                false.pure[F]
-              )
+              seesAgreement(first, second) &&^ seesAgreement(second, first) &&^ neverEventuallySeeDisagreement(
+                first,
+                second
+              ) &&^ neverEventuallySeeDisagreement(second, first)
```

Comment:
please also consider that the effects of all the conjuncts are going to be executed, which is a change to what's been the previous (lazy) semantics